src/main/java/frc/robot/subsystems/Shooter.java

```diff
@@ -156,12 +145,13 @@ public void feederMain() {
         if (oi.shootButton()) {
             startFeeder(feederSpeed);
             // startIntake(intakeSpeed);
-        } else if (oi.autoShootButton() && checkHoopVision()) { // Shoot when ready
+        } else if (oi.autoShootButton() && checkDependencies()) { // Shoot when ready
             if (Math.abs(vc.hoopr) <= vc.hoopChassisThreshold) { // Angle check
                 if (vc.hoopx <= vc.maxHoopDistance) // distance check
-                    if (oi.pilot.getLeftY() < 0.05) // Speed check (~0)
-                        if (Math.abs(getShooterRpms() - calculateShooterRPMS(vc.hoopx)) < vc.shooterThreshold) // flywheel rpm check
-                            startFeeder(feederSpeed);
+                    if (oi.pilot.getLeftY() < 0.05 && chassis.rpmToFps(chassis.getAverageRPM()) < 2) // Speed check
```

Comment:
Checking the actual chassis speed is good. Currently, getAverageRPM looks at all 4 wheels. The rear mecanum wheels will differ in speed from the front, especially if we're being pushed from the side by a defender or rotating into aiming position. So I wonder if we should only be looking at the front wheels. 2 RPM seems pretty low for the limit on wheel RPM, and also remember this is motor RPM so the chassis gear ratio should be factored in as well.

Reply:
I changed the average velocity but the check is for the chassis movement in feet per second

Reply:
👍